### PR TITLE
New configuration option: class_name_scalars

### PR DESCRIPTION
**Pull request by webimpress**
@weierophinney as we discussed, added new configuration option: `class_name_scalars`.

By default is set to `false`.

When it is set to `true` ConfigWriter will use `::class` notation on writing to configuration files.

In this PR also improvement some parts of the repository:
- Updated CS rules + CS fixes.
- Fixed invalid PHPDocs.
- Fixed syntax error in `ConfigResourceFactory` + tests.
- Added trailing comma in arrays.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
+.idea/
```
Comment (weierophinney) [inline .gitignore]: You should add this in your own global `.gitignore` rules, not in the project rules.

Comment (webimpress) [inline .gitignore]: In may project we have it so I thought it's ok

File: composer.json [hunk 2]
```diff
@@ -33,8 +33,9 @@
         "zendframework/zend-stdlib": "^2.7.7 || ^3.0.1"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8 || ^5.0",
-        "squizlabs/php_codesniffer": "^2.3.1"
+        "phpunit/phpunit": "^4.8 || ^5.5",
+        "squizlabs/php_codesniffer": "^2.6.2",
+        "zendframework/zend-servicemanager": "^2.7.6 || ^3.1.1"
```
Comment (weierophinney) [inline composer.json]: Why is this added? I'm not seeing it used in any of the new code. If anything, I'd expect to see container-interop as a dev dependency...

File: README.md [hunk 3]
```diff
@@ -71,6 +72,10 @@ The top-level configuration key for user configuration of this module is `zf-con
 Set this value to a boolean `true` if you want to use PHP 5.4's square bracket (aka "short") array
 syntax.
 
+#### Key: `class_name_scalars`
+
+Set this value to a boolean `true` if you want to use PHP 5.5's class name scalars (`::class` notation).
```
Comment (weierophinney) [inline README.md]: Add a bullet point:

```markdown
- Since 1.2.1
```
